### ISSUE TITLE
[layouts] ERC layout shows only one channel

### ISSUE THREAD

**Issue by vkz** [syl20bnr/spacemacs#4347]
#### Description
Even when you join multiple channels `spacemacs/persp-helm-mini` will only show one. Explicitly adding a buffer from another layer won't work.

#### Reproduction guide
- Create `@ERC` layout from custom layouts
- Join channel, say `#ocaml`
- Join another channel, say `#racket`
- `spacemacs/persp-helm-mini` will show only two buffers: `*scratch*` and `#racket`
- from layouts microstate try adding a buffer from another layer and choose `#ocaml` buffer
- `spacemacs/persp-helm-mini` will show only two buffers: `*scratch*` and `#ocaml`

#### System Info
- OS: darwin
- Emacs: 24.5.1
- Spacemacs: 0.105.0
- Spacemacs branch: develop (rev. 9642c35)
- Distribution: spacemacs
- Layers:
```elisp
(emoji erc command-log spacemacs-layouts eyebrowse clojure
       (auto-completion :variables auto-completion-return-key-behavior nil auto-completion-tab-key-behavior 'complete auto-completion-enable-help-tooltip nil auto-completion-enable-sort-by-usage t :disabled-for org erc)
       emacs-lisp
       (haskell :variables haskell-process-type 'stack-ghci haskell-enable-shm-support t haskell-enable-hindent-style "chris-done" haskell-enable-ghc-mod-support t)
       git github
       (shell :variables shell-default-shell 'eshell shell-default-position 'full shell-default-term-shell "/bin/bash" shell-enable-smart-eshell t)
       themes-megapack theming colors version-control restclient sexy key-seq)
```

**Comment by TheBB** [comment]
We should add a hook to `erc-mode` to add a buffer to the ERC perspective, if it exists.

**Comment by TheBB** [comment]
I assume adding buffers to the perspective does work. It's just that `#racket` wasn't added, so when you switched away from it, it won't show, either.

**Comment by AlejandroCatalina** [comment]
I just added a PR, I think it solves this case (it does for my case, RCIRC).

**Comment by syl20bnr** [comment]
Fixed in release 0.105.7!
